linkerd/client-policy/src/lib.rs

```diff
@@ -0,0 +1,225 @@
+#![deny(rust_2018_idioms, clippy::disallowed_methods, clippy::disallowed_types)]
+#![forbid(unsafe_code)]
+
+pub mod http;
+pub mod split;
+
+use linkerd_addr::NameAddr;
+pub use linkerd_policy_core::{meta, Meta};
+use once_cell::sync::Lazy;
+use std::{fmt, str::FromStr, sync::Arc};
+use tokio::sync::watch;
+
+#[derive(Clone, Debug)]
+pub struct Receiver {
+    inner: watch::Receiver<ClientPolicy>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ClientPolicy {
+    pub http_routes: Arc<[http::Route]>,
+    pub meta: Arc<Meta>,
+    pub backends: Vec<split::Backend>,
+}
```

Comment:
Echoing my comments on https://github.com/linkerd/linkerd2-proxy-api/pull/165#discussion_r1042661341, we're going to have to set this up so that HTTP is just one flavor of it.

Also, what does the `Meta` on this resource correspond to?  On the `Server`Policy` side, it indicates the `Server`. What does it mean here?

Reply:
> Echoing my comments on [linkerd/linkerd2-proxy-api#165 (comment)](https://github.com/linkerd/linkerd2-proxy-api/pull/165#discussion_r1042661341), we're going to have to set this up so that HTTP is just one flavor of it.

Yes, my expectation is that the `ClientPolicy` type would grow additional fields for e.g. `grpc_routes` and `tcp_routes` when support for other protocols are added to the `outbound` gRPC API.

> Also, what does the `Meta` on this resource correspond to? On the `ServerPolicy` side, it indicates the `Server`. What does it mean here?

Right now it's un-populated and should probably be removed. It might be desirable to have the name of the Service that the policy corresponds to, but that's already included in other metrics...I think the top-level metadata here ought to go away, at least until we decide how we want metrics for client policy to work?

Reply:
> Echoing my comments on [linkerd/linkerd2-proxy-api#165 (comment)](https://github.com/linkerd/linkerd2-proxy-api/pull/165#discussion_r1042661341), we're going to have to set this up so that HTTP is just one flavor of it.

Yes, my expectation is that the `ClientPolicy` type would grow additional fields for e.g. `grpc_routes` and `tcp_routes` when support for other protocols are added to the `outbound` gRPC API.

> Also, what does the `Meta` on this resource correspond to? On the `ServerPolicy` side, it indicates the `Server`. What does it mean here?

Right now it's un-populated and should probably be removed. It might be desirable to have the name of the Service that the policy corresponds to, but that's already included in other metrics...I think the top-level metadata here ought to go away, at least until we decide how we want metrics for client policy to work?

Reply:
Removed the `meta` field here in 9fe2284a8a86328c925bdccd1ea2c732ee61b766.